Code/autopkglib/URLGetter.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/python
+#
+# Copyright 2018 Michal Moravec
+# Based on code from Greg Neagle, Timothy Sutton and Per Olofsson
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""See docstring for URLGetter class"""
+
+import os.path
+import subprocess
+
+from autopkglib import Processor, ProcessorError, get_pref, log_err
+
+__all__ = ["URLGetter"]
+
+
+class URLGetter(Processor):
+    """Handles curl HTTP operatios. Server only as superclass. Not for direct use."""
+
+    description = __doc__
+
+    def curl_binary(self):
```

Comment:
`curl_binary()` method was copied from [autopkglib](https://github.com/autopkg/autopkg/blob/15c6307254d4ef1d1dbb8a2648a204a340f5b967/Code/autopkglib/__init__.py#L268.)

Reply:
In that case, why separate it out? For obvious reasons, it's completely guaranteed that autopkglib is present on the machine, so why not just use it? I'd rather deduplicate that entirely and just directly import the function from autopkglib.

Reply:
Given that statement, I'm going to review the code in here. To reduce complexity, we should consider changing the code in autopkglib to match the review I'm doing here.

Reply:
My intent is to remove `curl_cmd()` function from autopkglib (This was done in my original PR #446 https://github.com/autopkg/autopkg/pull/446/commits/7182befc54bcf1f310f837420053519e45d712bb).

Arguments:

1. All `curl` related code should be in `URLGetter`.
2. No one in autopkg org is using  `curl_cmd()` in their code therefore we should be able to remove it wihout any inpact.

Reply:
I'm sort of of the opposite opinion, that curl_cmd() should stay in autopkglib and be referenced here, but honestly don't see a real need to actually do that. So I'm satisfied with this methodology.